### PR TITLE
changed thp conditional to ansible_service_manager Fixes splunk/ansible-role-for-splunk#146

### PR DESCRIPTION
**Pull request by jewnix**
The current conditional did not work right.
The limit on RedHat <6 was because of the lack of `systemd`. Changing to using `ansible_service_manager` will do the same thing, and is more descriptive of the reason why it's done that way.

I moved it to the `configure_os.yml` so it can skip the whole `include_tasks.yml` altogether instead of going through the whole task and skippin if it does not meet the requirements.

### CODE REVIEW

File: roles/splunk/tasks/configure_os.yml [hunk 1]
```diff
@@ -12,7 +12,10 @@
 
 - name: Disable THP
   include_tasks: configure_thp.yml
-  when: "'full' in group_names"
+  when:
+    - ansible_service_mgr == "systemd"
```
Comment (anelson-splunk) [inline roles/splunk/tasks/configure_os.yml:16]: Where is `ansible_service_mgr` defined?

File: roles/splunk/tasks/configure_os.yml [hunk 2]
```diff
@@ -12,7 +12,10 @@
...
+    - ansible_connection != "docker"
+    - "'full' in group_names"
```
Comment (anelson-splunk) [inline roles/splunk/tasks/configure_os.yml:18]: Are the double quotes around this necessary?

File: roles/splunk/tasks/configure_os.yml (hunk 1 above)
Comment (dtwersky) [reply roles/splunk/tasks/configure_os.yml:16]: It's a built-in fact

File: roles/splunk/tasks/configure_os.yml (hunk 2 above)
Comment (dtwersky) [reply roles/splunk/tasks/configure_os.yml:18]: Yes, it needs the double quotes

**Review by dtwersky — APPROVED** [review]

**Pull request merged**